Source/Evolve/WarpXEvolve.cpp

```diff
@@ -924,12 +924,15 @@ WarpX::PushParticlesandDepose (int lev, amrex::Real cur_time, DtType a_dt_type,
         current_y = current_fp_nodal[lev][1].get();
         current_z = current_fp_nodal[lev][2].get();
     }
-    else if (WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay)
+#ifdef PSATD
+    else if (WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay and
+             electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD)
     {
         current_x = current_fp_vay[lev][0].get();
         current_y = current_fp_vay[lev][1].get();
         current_z = current_fp_vay[lev][2].get();
     }
+#endif
```

Comment:
Thanks, @RevathiJambunathan! 

If you want to leave the `#ifdef` directive, I think it will need to be `#ifdef WARPX_USE_PSATD` instead of `#ifdef PSATD` (that's why CI is currently failing, just tested locally).

However, I would have two requests/comments:

1) I would be more in favor of removing the `#ifdef` directive. Personally, I don't think it's needed here, as those few lines of code can be compiled even without PSATD.

2) I would argue that even the extra condition in the `if` statement, namely `electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD`, is not necessary here. Thanks to the assert message that you added in `WarpX::ReadParameters`, a situation where `WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay ` is true and `electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD` is false cannot occurr anymore. If your goal here was simply to make sure that the code clearly shows that Vay deposition and the "Vay current" exist only with the PSATD solver, I think we could add an inline comment stating `// Note: Vay deposition is supported only for PSATD, code currently aborts otherwise` or something like that. Otherwise, I think there are many other places in the code where a check on Vay deposition would need to be coupled to a check on the electromagnetic solver. But again, thanks to the assert message that you added in `WarpX::ReadParameters`, I think that's not necessary anymore: that single check takes care of all problems. What do you think?

Reply:
Thanks Edoardo! 
I agree that there would be too many places to add an additional if condition.
However, the way it read now, when debugging, it seemed as if its implemented for FDTD

With this PR, we have added comments in the documentation, an abort, and adding a comment makes it clear. So I agree with removing this ifdef and if condition altogether, but adding a meaningful comment.